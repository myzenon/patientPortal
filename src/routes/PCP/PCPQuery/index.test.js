import React from 'react'
import { PCPQuery } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
  const props = {
      patient: {
          ...mockObj.patient,
      },
      cmr: {
          ...mockObj.cmr,
          medication_000080833: {
              taken: "n",
              reasonNotTaken: {},
              reviewCompleted: true
          },
          medication_000293159: {
              taken: "n",
              reasonNotTaken: {},
              reviewCompleted: true
          },
          medication_000493990: {
              taken: "n",
              reasonNotTaken: {},
              reviewCompleted: true
          }
      },
      match: {
          params: {
              medId: '665820312'
          }
      },
  }

    const wrapper = render(
        <PCPQuery  { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks Next without changing the main doctor | go to meds identified', (done) => {
    const props = {
      cmr: {
          ...mockObj.cmr,
          medication_000080833: {
              taken: "n",
              reasonNotTaken: {},
              reviewCompleted: false
          },
          medication_000293159: {
              taken: "n",
              reasonNotTaken: {},
              reviewCompleted: false
          }
        },
        patient: {
            ...mockObj.patient,
        },  callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirPatientsIdPut')
            expect(get).toEqual({
                id: mockObj.patient.id
            })
            expect(post).toEqual({
                pcp: mockObj.patient.pcp,
                confirmedPcp: true,
                primaryPharmacy: mockObj.patient.primaryPharmacy,
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        setPatient: (patient) => {
            expect(patient).toEqual({
                ...mockObj.patient,
                pcp: mockObj.patient.pcp,
                confirmedPcp: true,
                primaryPharmacy: mockObj.patient.primaryPharmacy,
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/prepare/Chapter2')
            done()
        }
    }
    const wrapper = mount(
        <PCPQuery  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').first().simulate('click')
    }, 0)
})

test('clicks Next with changing the main doctor (9283202372) | go to method delivery', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: false
            },
            medication_000293159: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: false
            }
          },
        patient: {
            ...mockObj.patient,
        },  callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirPatientsIdPut')
            expect(get).toEqual({
                id: mockObj.patient.id
            })
            expect(post).toEqual({
                confirmedPcp: true,
                confirmedPharmacy: true,
                pcp: "9283202372",
                primaryPharmacy: "1568477834"
            })
            return Promise.resolve()
        },
        setPatient: (patient) => {
            expect(patient).toEqual({
                ...mockObj.patient,
                confirmedPcp: true,
                confirmedPharmacy: true,
                pcp: "9283202372",
                primaryPharmacy: "1568477834"
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/prepare/Chapter2')
            done()
        }
    }
    const wrapper = mount(
        <PCPQuery  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('.choiceWrapper').first().simulate('click')
        wrapper.find('.choiceWrapper').at(2).simulate('click')
        wrapper.find('button').first().simulate('click')
    }, 0)
})

test('clicks Next without changing the main doctor | go to meds identified', (done) => {
    const props = {
      cmr: {
          ...mockObj.cmr,
          medication_000080833: {
              taken: "n",
              reasonNotTaken: {},
              reviewCompleted: false
          },
          medication_000293159: {
              taken: "n",
              reasonNotTaken: {},
              reviewCompleted: false
          }
        },
        patient: {
            ...mockObj.patient,
        },  callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirPatientsIdPut')
            expect(get).toEqual({
                id: mockObj.patient.id
            })
            expect(post).toEqual({
                pcp: mockObj.patient.pcp,
                confirmedPcp: true,
                primaryPharmacy: mockObj.patient.primaryPharmacy,
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        setPatient: (patient) => {
            expect(patient).toEqual({
                ...mockObj.patient,
                pcp: mockObj.patient.pcp,
                confirmedPcp: true,
                primaryPharmacy: mockObj.patient.primaryPharmacy,
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/prepare/Chapter2')
            done()
        }
    }
    const wrapper = mount(
        <PCPQuery  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').first().simulate('click')
    }, 0)
})

test('clicks Next with selecting other choice | go to method delivery', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: false
            },
            medication_000293159: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: false
            }
          },
        patient: {
            ...mockObj.patient,
        },  callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirPatientsIdPut')
            expect(get).toEqual({
                id: mockObj.patient.id
            })
            expect(post).toEqual({
                pcp: "otherPractitioner",
                otherPractitioner: "",
                confirmedPcp: true,
                primaryPharmacy: "otherPharmacy",
                otherPharmacy: "",
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        setPatient: (patient) => {
            expect(patient).toEqual({
                ...mockObj.patient,
                pcp: "otherPractitioner",
                otherPractitioner: "",
                confirmedPcp: true,
                primaryPharmacy: "otherPharmacy",
                otherPharmacy: "",
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/prepare/Chapter2')
            done()
        }
    }
    const wrapper = mount(
        <PCPQuery  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('.choiceWrapper').at(1).simulate('click')
        wrapper.find('.choiceWrapper').at(3).simulate('click')
        wrapper.find('button').first().simulate('click')
    }, 0)
})

test('clicks Next with selecting other choice and fill in other text field | go to method delivery', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: false
            },
            medication_000293159: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: false
            }
          },
        patient: {
            ...mockObj.patient,
        },  callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirPatientsIdPut')
            expect(get).toEqual({
                id: mockObj.patient.id
            })
            expect(post).toEqual({
                pcp: "otherPractitioner",
                otherPractitioner: "Dr. Adam Saunter",
                confirmedPcp: true,
                primaryPharmacy: "otherPharmacy",
                otherPharmacy: "44 Shirley Ave Gastonia, NC, USA 28052",
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        setPatient: (patient) => {
            expect(patient).toEqual({
                ...mockObj.patient,
                pcp: "otherPractitioner",
                otherPractitioner: "Dr. Adam Saunter",
                confirmedPcp: true,
                primaryPharmacy: "otherPharmacy",
                otherPharmacy: "44 Shirley Ave Gastonia, NC, USA 28052",
                confirmedPharmacy: true
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/prepare/Chapter2')
            done()
        }
    }
    const wrapper = mount(
        <PCPQuery  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('.choiceWrapper').at(1).simulate('click')
        wrapper.find('input[name="otherPractitioner"]').simulate('change', {
            target: {
                name: 'otherPractitioner',
                value: 'Dr. Adam Saunter'
            }
        })
        wrapper.find('.choiceWrapper').at(3).simulate('click')
        wrapper.find('input[name="otherPharmacy"]').simulate('change', {
            target: {
                name: 'otherPharmacy',
                value: '44 Shirley Ave Gastonia, NC, USA 28052'
            }
        })
        wrapper.find('button').first().simulate('click')
    }, 0)
})
