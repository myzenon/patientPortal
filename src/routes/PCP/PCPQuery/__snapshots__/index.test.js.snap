// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <p>
      Please select
    </p>
    <div
      class="flexContainer"
    >
      <div
        class="doctorSelectContainer"
      >
        <div
          class="title"
        >
          Your main doctor:
        </div>
        <div
          class="selectWrapper"
        >
          <div
            class="activeChoiceWrapper"
          >
            <div
              class="radioWrapper"
            >
              <div
                class="activeRadio"
              />
            </div>
            <div
              class="text"
            >
              Dr.  Saunter Adam - 44 Shirley Ave Gastonia , NC , USA. 28052
            </div>
          </div>
          <div
            class="choiceWrapper"
          >
            <div
              class="radioWrapper"
            >
              <div
                class="radio"
              />
            </div>
            <div
              class="text"
            >
              Dr.  Mahoney Sandra - 14 Bowman St. Monsey , NY , USA. 10952
            </div>
          </div>
          <div
            class="choiceWrapper"
          >
            <div
              class="radioWrapper"
            >
              <div
                class="radio"
              />
            </div>
            <div
              class="text"
            >
              Other
              <input
                class="inputText"
                name="otherPractitioner"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="pharmacySelectContainer"
      >
        <div
          class="title"
        >
          Your preferred pharmacy:
        </div>
        <div
          class="selectWrapper"
        >
          <div
            class="activeChoiceWrapper"
          >
            <div
              class="radioWrapper"
            >
              <div
                class="activeRadio"
              />
            </div>
            <div
              class="text"
            >
              Walgreens Pharmacy - 3601 California St, 3601 California Street, San Francisco, CA 94118-1701
            </div>
          </div>
          <div
            class="choiceWrapper"
          >
            <div
              class="radioWrapper"
            >
              <div
                class="radio"
              />
            </div>
            <div
              class="text"
            >
              Walgreens Pharmacy - 3838 California St, San Francisco, CA 94118-1522
            </div>
          </div>
          <div
            class="choiceWrapper"
          >
            <div
              class="radioWrapper"
            >
              <div
                class="radio"
              />
            </div>
            <div
              class="text"
            >
              Other
              <input
                class="inputText"
                name="otherPharmacy"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="button btn btn-secondary"
      type="button"
    >
      <b>
        NEXT
      </b>
    </button>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
